tests/Agency/StoreTestAPI.cpp

```diff
@@ -0,0 +1,977 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Ignacio Rodriguez
+/// @author Max Neunhoeffer
+/// @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+#include "gtest/gtest.h"
+
+#include "Agency/Store.h"
+#include "Mocks/Servers.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Compare.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include <random>
+
+namespace arangodb {
+namespace tests {
+namespace store_test_api {
+
+class StoreTestAPI : public ::testing::Test {
+ public:
+  StoreTestAPI() : _server(), _store(_server.server(), nullptr) {
+  }
+ protected:  
+  arangodb::tests::mocks::MockCoordinator _server;
+  arangodb::consensus::Store _store;
+
+  auto read(std::string const &json) 
+  {
+    try {
+      auto q {VPackParser::fromJson(json)};
+      auto result {std::make_shared<VPackBuilder>()};
```

Comment:
Here, it is totally fine (and even preferable), since one can see that the type is a shared_ptr to a VPackBuilder.